source/features/reactions-avatars.tsx

```diff
@@ -21,13 +21,33 @@ type Participant = {
 };
 
 function getParticipants(button: HTMLButtonElement): Participant[] {
-	// The list of people who commented is in an adjacent `<tool-tip>` element #5698
-	const users = button.nextElementSibling!
-		.textContent
-		.replace(/ reacted with.*/, '')
-		.replace(/,? and /, ', ')
-		.replace(/, \d+ more/, '')
-		.split(', ');
+	let users;
+
+	if (button.getAttribute('role') === 'switch') {
+		users = button.getAttribute('aria-label')!
+			.replace(/.*including /, '')
+			.replace(/\)/, '')
+			.replace(/,? and /, ', ')
+			.replace(/, \d+ more/, '')
+			.split(', ');
+	} else if (button.nextElementSibling) {
+		// TODO: Remove after March 2024
+		// The list of people who commented is in an adjacent `<tool-tip>` element #5698
+		const tagName = button.nextElementSibling.tagName;
+		if (tagName !== 'TOOL-TIP')
+			return [];
```

Comment:
This needs to be cleaned up a bit.

- In what case is it not `tool-tip`? Is the selector somewhere selecting the wrong elements?
- Does the current version work correctly on discussions?

At a minimum the code should be changed to:

```suggestion
		// TODO: Remove after March 2024
		// The list of people who commented is in an adjacent `<tool-tip>` element #5698
	} else if (button.nextElementSibling?.tagName === 'TOOL-TIP') {
```

Reply:
How about the questions?

Reply:
> Does the current version work correctly on discussions?

It works on discusssions.

![CleanShot 2024-10-24 at 08 39 35@2x](https://github.com/user-attachments/assets/343d08ec-96d2-48cf-9a5f-65b84f30b717)

Reply:
> In what case is it not tool-tip? Is the selector somewhere selecting the wrong elements?

As you can see, we get the button then we get its `nextElementSibling`.

https://github.com/refined-github/refined-github/blob/704e344b04e8e758637cdbc74a333f94169e9b1a/source/features/reactions-avatars.tsx#L23

I find the reason it's the `button.nextElementSibling` is a button, but the comment says `The list of people who commented is in an adjacent <tool-tip> element`. So I think we should to determine.

<img width="1518" alt="image" src="https://github.com/user-attachments/assets/4c1429c4-7cb7-445d-9a37-ddafe8d61730">

If `button.nextElementSibling` is a button, this is what's happened.

```
❌ Refined GitHub: reactions-avatars
errors.js:77 📕 0.0.0 → SyntaxError: Failed to execute 'querySelector' on 'Document': '[href="/    🚀
            3
" i] img' is not a valid selector.
    at $ (select-dom.js:8:38)
    at getUserAvatar (get-user-avatar.js:8:25)
    at getParticipants (reactions-avatars.js:53:20)
    at reactions-avatars.js:86:19
    at Array.map (<anonymous>)
    at showAvatarsOn (reactions-avatars.js:86:5)
    at IntersectionObserver.rootMargin (reactions-avatars.js:65:4)
```

Reply:
```tsx
if (button.getAttribute('disabled') === 'disabled')
		return [];
```

Maybe we just should add this.

Reply:
👆 exactly. If the tooltip doesn't exist when it's disabled, then the observed selector should just be `:not([disabled])`, getParticipants shouldn't be called at all